### PR TITLE
Civetweb compilation fixes

### PR DESCRIPTION
**Pull request by jukkar**
Some issues with civetweb when `CONFIG_NET_TCP_ISN_RFC6528` is enabled.

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: samples/net/civetweb/http_server/prj.conf [hunk 1]
```diff
@@ -30,3 +30,7 @@ CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.0.2.2"
 
 # logging
 CONFIG_NET_LOG=y
+
+# Do not include mbedtls via this option as civetweb does not
+# work properly with mbedtls.
+CONFIG_NET_TCP_ISN_RFC6528=n
```
Comment (pabigot) [inline samples/net/civetweb/http_server/prj.conf:36]: Could this be checked by a `depends on !CONFIG_NET_TCP_ISN_RFC6528` in the `CIVETWEB` Kconfig declaration, so this isn't only checked for these two samples?

**Review by pabigot — COMMENTED** [review]
I often get bit when trying to use features because there are dependencies that aren't supported, but they aren't checked.  Is there a way to handle this without only doing it in the samples?

**Review by pfalcon — APPROVED** [review]
Confirming fixes CI breakage, please merge.

File: samples/net/civetweb/http_server/prj.conf (hunk 1 above)
Comment (jukkar) [reply samples/net/civetweb/http_server/prj.conf:36]: > Could this be checked by a `depends on !CONFIG_NET_TCP_ISN_RFC6528` in the `CIVETWEB` Kconfig declaration

Sure, I can send a separate PR for that.

Comment (jukkar) [reply samples/net/civetweb/http_server/prj.conf:36]: See #33215

**Pull request merged**